app/controllers/events_controller.rb

```diff
@@ -0,0 +1,92 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :destroy, :attend]
+  #before_action :correct_user, only: %i[edit update]
+
+  # GET /events
+  # GET /events.json
+  def index
+    if params[:events].nil? || params[:events][:type] == 'upcoming'
+      @events = Event.upcoming.paginate(page: params[:page])
+      @header = 'Upcoming Events'
+    else
+      @events = Event.prev.paginate(page: params[:page])
+      @header = 'Previous Events'
+    end
+  end
+
+  # GET /events/1
+  # GET /events/1.json
+  def show
+    @event = Event.find(params[:id])
+    @event_attendees = @event.attendees
+    #@going = @event.attendees.where('accepted AND NOT declined')
+    #@not_going = @event.attendees.where('NOT accepted AND declined')
+  end
+
+  # GET /events/new
+  def new
+    @event = Event.new
+  end
+
+  # GET /events/1/edit
+  def edit
+    session[:event_id] = params[:id]
+    @event = Event.find(params[:id])
+    @users = User.where('id != ?', current_user.id)
+  end
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = Event.new(event_params.merge(user_id: current_user.id))
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to @event, notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
+      else
+        format.html { render :new }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
```

Comment:
I see that you use the scaffold to generate the actions but it is better if you create the methods manually because with the scaffold you have a lot of code that you don't need.